components/ca-select.js

```diff
@@ -0,0 +1,443 @@
+define([
+    './helpers/component-support.js',
+    './helpers/create-element.js',
+    './helpers/types.js',
+    './helpers/add-event.js',
+    './helpers/cancel-event.js',
+    './helpers/parent-by-attribute.js',
+    './helpers/trace.js',
+    'document-register-element'
+], (componentSupport, createElement, types, addEvent, cancelEvent, parentByAttribute, trace) => {
+
+    /**
+     * Select class for Select web component
+     * @extends HTMLElement
+     */
+    class Select extends HTMLElement {
+
+        /**
+        * Called when element is initialised
+        * @returns {void} void
+        */
+        createdCallback() {
+
+            const self = this;
+            this.isTouch = ('ontouchend' in document.documentElement);
+            this.selections = {};
+
+            switch (self.style) {
+
+                case 'checkbox':
+                    this.root = this;
+                    break;
+
+                default:
+                    this.root = createElement(this, 'select', {});
+                    this.root.multiple = this.multiple;
+                    this.root.onchange = this.onChangeHandler.bind(self);
```

Comment:
That doesn't sound right.  this.onChangeHandler() would invoke it.

I think it should be

    this.root.onchange = this.onChangeHandler.bind(this);

or

    this.root.onchange = () => this.onChangeHandler();